pkg/cmd/cli/cmd/generate.go

```diff
@@ -0,0 +1,99 @@
+package cmd
+
+import (
+	"context"
+
+	"github.com/containers/image/v5/transports/alltransports"
+	"github.com/containers/image/v5/types"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/source-to-image/pkg/api"
+	"github.com/openshift/source-to-image/pkg/api/constants"
+	"github.com/openshift/source-to-image/pkg/build/strategies/dockerfile"
+	"github.com/openshift/source-to-image/pkg/util/fs"
+)
+
+// getImageLabels attempts to inspect an image existing in a remote registry.
+func getImageLabels(ctx context.Context, imageName string) (map[string]string, error) {
+	ref, err := alltransports.ParseImageName(imageName)
+	if err != nil {
+		return nil, err
+	}
+
+	img, err := ref.NewImage(ctx, &types.SystemContext{})
+	if err != nil {
+		return nil, err
+	}
+
+	imageMetadata, err := img.Inspect(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return imageMetadata.Labels, nil
+}
+
+// generateDockerfile generates a Dockerfile with the given configuration.
+func generateDockerfile(cfg *api.Config) error {
+	fileSystem := fs.NewFileSystem()
+	builder, err := dockerfile.New(cfg, fileSystem)
+	if err != nil {
+		return err
+	}
+
+	_, err = builder.Build(cfg)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// adjustConfigWithImageLabels adjusts the configuration with given labels.
+func adjustConfigWithImageLabels(cfg *api.Config, labels map[string]string) {
+	if v, ok := labels[constants.ScriptsURLLabel]; ok {
+		cfg.ScriptsURL = v
+	}
+
+	if v, ok := labels[constants.DestinationLabel]; ok {
+		cfg.Destination = v
+	}
+
+}
+
+// NewCmdGenerate implements the S2I cli generate command.
+func NewCmdGenerate(cfg *api.Config) *cobra.Command {
+	generateCmd := &cobra.Command{
+		Use:   "generate <image> <dockerfile>",
+		Short: "Generate a Dockerfile based on the provided builder image",
+		Example: `
+# Generate a Dockerfile from a remote builder image:
+$ s2i generate docker://quay.io/redhat-developer/app-binding-operator Dockerfile.gen
```

Comment:
I'd use one of our canonical s2i images as an example - `docker.io/centos/nodejs-10-centos7`

Reply:
@isutton Could you please make the above change?

Reply:
@isutton please update the example here.

Reply:
Also a bit of a nit - is it worth switching the order around? Thoughts on the following semantics?

```
$ s2i generate DOCKERFILE [IMAGE]
```

or 

```
$ s2i generate DOCKERFILE --from=[IMAGE]
```

Reply:
From what I understood, IMAGE is always required in this use case since there's no information regarding that; additionally, the protocol currently must be informed by the user to refer to the image since `containers/image` is responsible for retrieving its contents and inspecting its labels.

I believe the following example is closer to what we want, where DOCKERFILE could have the `Dockerfile` value by default:

```
$ s2i generate IMAGE [DOCKERFILE]
```

During implementation, I opted for DOCKERFILE to be mandatory and not having defaults to avoid unwillingly Dockerfile overwrites.
---
The requested example modification has been made.

Reply:
Right, IMAGE is always required since it is the s2i builder image. I think Adam is OK with that.

I do like `s2i generate DOCKERFILE --from=quay.io/builder/image`

 instead of 

`s2i generate quay.io/builder/image Dockerfile.gen`

Reply:
> From what I understood, IMAGE is always required in this use case since there's no information regarding that; 

Correct

> During implementation, I opted for DOCKERFILE to be mandatory and not having defaults to avoid unwillingly Dockerfile overwrites.

I agree with this choice for now. We can always opt to make it optional later; I have other ideas on how to arrange the arguments to this command that are worth discussing in a follow-up PR.